wallet/wallet.c

```diff
@@ -434,12 +417,14 @@ static bool wallet_stmt2channel(struct wallet *w, sqlite3_stmt *stmt,
 	chan->peer->next_index[REMOTE] = sqlite3_column_int64(stmt, col++);
 	chan->peer->next_htlc_id = sqlite3_column_int64(stmt, col++);
 
-	if (sqlite3_column_hexval(stmt, col++, &temphash, sizeof(temphash))) {
+	if (sqlite3_column_type(stmt, col) != SQLITE_NULL) {
+		assert(sqlite3_column_bytes(stmt, col) == 32);
 		chan->peer->funding_txid = tal(chan->peer, struct sha256_double);
-		*chan->peer->funding_txid = temphash;
+		memcpy(chan->peer->funding_txid, sqlite3_column_blob(stmt, col), 32);
```

Comment:
I wonder if we should add a helper whch does the size assert for us?  eg. sqlite3_copy_val(stmt, col, dest, len)?  Seems to be a common pattern (which sqlite3_column_hexval() did for us).